src/ui/control/geolocate_control.js

```diff
@@ -137,7 +137,31 @@ class GeolocateControl extends Evented {
         this._map = (undefined: any);
     }
 
+    _isOutOfMapMaxBounds(position: Position) {
+        const bounds = this._map.getMaxBounds();
+        const coordinates = position.coords;
+
+        return bounds && (
+            coordinates.longitude < bounds.getWest() ||
+            coordinates.longitude > bounds.getEast() ||
+            coordinates.latitude < bounds.getSouth() ||
+            coordinates.latitude > bounds.getNorth()
+        );
+    }
+
     _onSuccess(position: Position) {
+        if (this._isOutOfMapMaxBounds(position)) {
+            this._watchState = 'ACTIVE_ERROR';
```

Comment:
per what we do at
https://github.com/mapbox/mapbox-gl-js/blob/87cac3be79880fe6b6cf532baa023c52b112cf1f/src/ui/control/geolocate_control.js#L221-L245

we should only set the ACTIVE_ERROR state when the map is in active tracking, if you're in background tracking and then you go outside the maxBounds then we should leave the error icon as a background error and not active error.

Reply:
You're right. Is it fine to add another class for out of bounds poistion? Something like `mapboxgl-ctrl-geolocate-out-of-bounds`?

Reply:
Unless you're proposing a new style for the button, then I'd say stick with the current classes.